schema/queue/message/impl/RabbitMqMessageRouter.go

```diff
@@ -173,18 +177,23 @@ func (cmr *CommonMessageRouter) getSender(pin string) *CommonMessageSender {
 	}
 }
 
-func (cmr *CommonMessageRouter) getMessageGroupWithPins(queue map[string]configuration.QueueConfig, message *p_buff.MessageGroupBatch) map[string]*p_buff.MessageGroupBatch {
+func (cmr *CommonMessageRouter) getMessageGroupWithPins(queue map[string]configuration.QueueConfig, msgGrBatch *p_buff.MessageGroupBatch) map[string]*p_buff.MessageGroupBatch {
 	//Here will be added filter handling
 	result := make(map[string]*p_buff.MessageGroupBatch)
-	for pin, _ := range queue {
-
-		msgBatch := p_buff.MessageGroupBatch{}
-		for _, messageGroup := range message.Groups {
-			//doing filtering based on queue filters on message_group
-			msgBatch.Groups = append(msgBatch.Groups, messageGroup)
+	for pin, config := range queue {
+		if len(config.Filters) > 0 {
+			msgBatch := p_buff.MessageGroupBatch{}
+			for _, messageGroup := range msgGrBatch.Groups {
+				//doing filtering based on queue filters on message_group
+				if cmr.filterStrategy.Verify(messageGroup, config.Filters) {
+					msgBatch.Groups = append(msgBatch.Groups, messageGroup)
+				}
+			}
+			result[pin] = &msgBatch
+		} else {
+			result[pin] = msgGrBatch
```

Comment:
We should change the behavior for filters because repacking batches is very expensive. If the whole batch matches the filter we publish it to this pin. Otherwise, we do not.

Also, I think it should be moved inside the `SendAll` function to avoid `map` allocation